### PR TITLE
feat: [CDS-38805]: aws infra def

### PR DESCRIPTION
**Pull request by azharness**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
trigger rebase

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+  const formikRef = useRef<FormikProps<SshWinRmAwsInfrastructureUI> | null>(null)
+  const delayedOnUpdate = useRef(debounce(onUpdate || noop, 300)).current
+
+  const { expressions } = useVariablesExpression()
+  const { getString } = useStrings()
+
+  const [regions, setRegions] = useState<SelectOption[]>([])
+
+  const [tags, setTags] = useState<SelectOption[]>([])
+  const [canTagsHaveFixedValue, setCanTagsHaveFixedValue] = useState(
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx:109]: Can add comments in the code around the logic of this function

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+    getMultiTypeFromValue(initialValues.region) === MultiTypeInputType.FIXED &&
+      getMultiTypeFromValue(initialValues.connectorRef) === MultiTypeInputType.FIXED
+  )
+
+  const parsedInitialValues = useMemo(() => {
+    const initials = {
+      ...initialValues
+    }
+    if (initialValues.credentialsRef) {
+      set(initials, 'sshKey', get(initialValues, 'credentialsRef', ''))
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx:119]: Why are there 2 different names for the same thing, : sshKey and credentialsRef

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+                  <Layout.Horizontal>
+                    <FormInput.MultiSelectTypeInput
+                      name="tags"
+                      label={getString('tagLabel')}
+                      selectItems={tags}
+                      className={`tags-select ${css.inputWidth}`}
+                      placeholder={isTagsLoading ? getString('loading') : getString('tagsLabel')}
+                      multiSelectTypeInputProps={{
+                        placeholder: isTagsLoading ? getString('loading') : getString('tagsLabel'),
+                        onFocus: () => {
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx:345]: If we have connector and regions by default and we fetch tags on mount of the component and after that the person focuses this element the API call is made again , hence the first call is redundant.

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+    awsInstanceFilter: { tags: {}, vpcs: [] },
+    connectorRef: '',
+    credentialsRef: '',
+    region: ''
+  }
+
+  /* istanbul ignore next */
+  protected stepIcon: IconName = 'service-aws'
+  /* istanbul ignore next */
+  protected stepName = 'Specify your AWS Infrastructure'
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx:416]: useStrings

**Review by vikrant-harness — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]
We have skipped some things here in this PR - like validations, form reset issue etc.
but these I have handled in my pr

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx (hunk 2 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/PipelineSteps/SshWinRmAwsInfrastructureSpec/SshWinRmAwsInfrastructureSpec.tsx:119]: @azharness I think we should use credentialsRef everywhere

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**